src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,219 @@
+        User secondUser = userConstructor(STANDARD_LOGIN_OK,
+                PASSWORD_OK,
+                STANDARD_AGE_OK,
+                STANDARD_ID_OK);
+        Storage.people.add(standartUser);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(secondUser),
+                "User already exist");
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        standartUser = null;
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "You can't register null user");
+    }
+
+    @Test
+    void register_user_specialCharacterLogin_notOk() {
+        standartUser.setLogin(SPECIAL_CHARACTERS);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "You can't register login with special characters, use only letters and numbers");
+    }
+
+    @Test
+    void register_user_NullLogin_notOk() {
+        standartUser.setLogin(null);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "Username can't be null");
+    }
+
+    @Test
+    void register_user_NullPassword_notOk() {
+        standartUser.setPassword(null);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "Password can't be null");
+    }
+
+    @Test
+    void register_user_ageLess18_notOk() {
+        standartUser.setAge(AGE_LESS);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "Age can't be less that " + AGE_LESS);
+    }
+
+    @Test
+    void register_user_ageBiggest_notOk() {
+        standartUser.setAge(AGE_BIGGEST);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "The age can't be so great");
+    }
+
+    @Test
+    void register_user_negativeId_notOk() {
+        standartUser.setId(NEGATIVE_ID);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "The id can't be negative");
+    }
+
+    @Test
+    void register_user_emptyLogin_notOk() {
+        standartUser.setLogin(EMPTY);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "Login can't be empty");
+    }
+
+    @Test
+    void register_user_emptyPassword_notOk() {
+        standartUser.setPassword(EMPTY);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "Login can't be empty");
+    }
+
+    @Test
+    void equals_nullObject_notOk() {
+        assertNotEquals(null, standartUser, "User can't equal null");
+    }
```

Comment:
redundant test. Actually here you are testing your test class :)

Reply:
Just in case)